product_brand_stock/models/stock_move.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Francesco Apruzzese <cescoap@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    product_brand_id = fields.Many2one(
+        comodel_name="product.brand",
+        related="product_id.product_brand_id",
+        string="Brand",
+        store=True,
```

Comment:
we need a pre_init_hook to initialize this column

Reply:
yes, sure, but for a production database with a substantial number of stock_move rows, the initialization will take an extensive amount of time.

in a pre_init_hook create the column and initialize it with one sql query

```sql
ALTER TABLE stock_move ADD COLUMN IF NOT EXISTS product_brand_id integer;
UPDATE stock_move m SET product_brand_id=p.product_brand_id FROM product p where p.id=m.product_id;
```

Reply:
> yes, sure, but for a production database with a substantial number of stock_move rows, the initialization will take an extensive amount of time.
> 
> in a pre_init_hook create the column and initialize it with one sql query
> 
> ```sql
> ALTER TABLE stock_move ADD COLUMN IF NOT EXISTS product_brand_id integer;
> UPDATE stock_move m SET product_brand_id=p.product_brand_id FROM product p where p.id=m.product_id;
> ```

I think it's an over-engineered request, but I made it anyway.
The query however is more complex because the brand is on the template and not the product.
Let me know if it is okay so I can replicate it on the other PR.

Reply:
Thank you for doing it anyway,

In other OCA repos, we you are used to this init hooks for large tables when adding a computed & stored field.